packages/shared/src/components/FeedItemComponent.tsx

```diff
@@ -170,12 +185,18 @@ export default function FeedItemComponent({
     ranking,
   );
 
-  const { shouldUseMobileFeedLayout } = useFeedLayout();
-  const { PostTag, AdTag, PlaceholderTag, MarketingCtaTag } = getTags(
-    isList,
-    shouldUseMobileFeedLayout,
-    (item as PostItem).post?.type,
-  );
+  const { shouldUseListFeedLayout, isListModeV1, shouldUseListModeV1 } =
+    useFeedLayout();
+  const { PostTag, AdTag, PlaceholderTag, MarketingCtaTag } = getTags({
+    isHorizontal,
+    isList: isListProp,
+    isListFeedLayout: shouldUseListFeedLayout,
+    shouldUseListModeV1,
+    postType: (item as PostItem).post?.type,
+  });
+
+  const insaneMode = isListModeV1 || isHorizontal ? false : insaneModeProp;
+  const isList = isListModeV1 || isHorizontal ? false : isListProp;
```

Comment:
Not the cleanest naming i'd say, finding it hard to envision what either does (I know it's mostly legacy, also non blocking :))

Reply:
I know, just wanted to keep the same namings used around at the moment, if this wins we have quite a lot of clean up to do, isList/insaneMode won't be in so many places it is anymore. So just thinking is best to do then.